checkov/terraform/checks/resource/base_resource_check.py

```diff
@@ -28,6 +29,21 @@ def __init__(
         self.supported_resources = supported_resources
         resource_registry.register(self)
 
+    @staticmethod
+    def _is_variable_dependant(value: Any) -> bool:
+        if not isinstance(value, str):
+            return False
+
+        if value.startswith('var.') or value.startswith('local.') or value.startswith('module.'):
+            return True
+
+        if "${" not in value:
+            return False
+
+        if find_var_blocks(value):
```

Comment:
if I'm not mistaken, then this should cover the `value.startswith('var.')` case

Reply:
We need to check for `var.` before filtering out strings with no interpolation

Reply:
We need to check for `var.` before filtering out strings that do not have interpolation

Reply:
👍